File: BEMLIB_22.10/01_num_meth/06_interp_diff/lagrange.m
function yint = lagrange (N,x,y,xint)

%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
% perform Lagrange interpolation
% on a specified set of N+1 points
%
% (x, y): data
% xint: interpolated x value
%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

 yint = 0.0;

 for i=1:N+1
    lpoly = 1.0;
    for j=1:N+1
      if (j~=i)
         lpoly = lpoly*(xint-x(j))/(x(i)-x(j));
      end
    end
    yint = yint + lpoly*y(i);
 end

%----
% done
%----

return

File: BEMLIB_22.10/01_num_meth/06_interp_diff/lagrange.f
      subroutine lagrange
     +
     +   (xint
     +   ,n
     +   ,x
     +   ,f
     +   ,fint
     +   )

c-----------------------------------------
c Copyright by C. Pozrikidis, 1999
c All rights reserved.
c
c This program is to be used only under the
c stipulations of the licensing agreement.
c----------------------------------------

c------------------------------------------------
c This program accompanies the book:
c
c C. Pozrikidis
c ``Numerical Computation in Science and Engineering''
c Oxford University Press, 1998
c------------------------------------------------

c------------------------
c  Lagrange interpolation
c------------------------

      Implicit Double Precision (a-h,o-z)

      Dimension x(50),f(50)

      n1 = n+1

c----------------------
c Evaluation of the sum
c----------------------

      sum = 0.0D0

      Do i=1,n1
        prod = 1.0D0
        Do j=1,n1
          If(j.ne.i) prod = prod*(xint-x(j))/(x(i)-x(j))
        End Do
        sum = sum + prod*f(i)
      End Do
	
      fint = sum

c-----
c Done
c-----

      Return
      End
